spateo/tdr/models/models_backbone/backbone_methods.py

```diff
@@ -61,17 +63,23 @@ def fit(self, data: np.ndarray, epochs: int = 500, nodes: int = 25, lr: float =
         """
         num_dim = data.shape[1]  # get number of dimensions for pts
 
-        # create models, base and intermediate
-        model = self.create_model(num_dim, nodes=nodes, lr=lr)
-        bname = model.layers[2].name  # bottle-neck layer name
+        # create model
+        self.model = self.create_model(num_dim, nodes=nodes)
+        optimizer = optim.Adam(self.model.parameters(), lr=lr)
+
+        data_tensor = torch.tensor(data, dtype=torch.float32)
 
-        # The itermediate model gets the output of the bottleneck layer,
-        # which acts as the projection layer.
-        self.intermediate_layer_model = Model(inputs=model.input, outputs=model.get_layer(bname).output)
```

Comment:
I implemented `intermediate_layer_model` inside the AutoEncoder later on, which should really be removed in self, I need to test it a bit further

Reply:
I have tested this function on simulated data, and it runs well. However, I haven't found specific use cases for this function. Could someone provide specific scenarios where this function is used so that I can further optimize it?